external_data/task_functions.py

```diff
@@ -0,0 +1,534 @@
+            "_id": row["siren"],
+            "doc": {
+                "liste_finess": list(set(row["liste_finess"])),
+            },
+        }
+
+
+def generate_updates_nondiff(df, current_color):
+    for index, row in df.iterrows():
+        yield {
+            "_op_type": "update",
+            "_index": "siren-" + current_color,
+            "_type": "_doc",
+            "_id": row["siren"],
+            "doc": {
+                "is_nondiffusible": True,
+            },
+        }
+
+
+def generate_updates_rge(df, current_color):
+    df["liste_rge"] = df["liste_rge"].apply(literal_eval)
+    for index, row in df.iterrows():
+        yield {
+            "_op_type": "update",
+            "_index": "siren-" + current_color,
+            "_type": "_doc",
+            "_id": row["siren"],
+            "doc": {
+                "liste_rge": list(set(row["liste_rge"])),
+            },
+        }
+
+
+def generate_updates_spectacle(df, current_color):
+    for index, row in df.iterrows():
+        yield {
+            "_op_type": "update",
+            "_index": "siren-" + current_color,
+            "_type": "_doc",
+            "_id": row["siren"],
+            "doc": {
+                "is_entrepreneur_spectacle": True,
+            },
+        }
+
+
+def generate_updates_uai(df, current_color):
+    df["liste_uai"] = df["liste_uai"].apply(literal_eval)
+    for index, row in df.iterrows():
+        yield {
+            "_op_type": "update",
+            "_index": "siren-" + current_color,
+            "_type": "_doc",
+            "_id": row["siren"],
+            "doc": {
+                "liste_uai": list(set(row["liste_uai"])),
+            },
+        }
+
+
+def update_es(
+    type_file,
+    new_file,
+    error_file,
+    select_color,
+    **kwargs,
+) -> None:
+    if select_color == "current":
+        color = kwargs["ti"].xcom_pull(key="current_color", task_ids="get_colors")
+    if select_color == "next":
+        color = kwargs["ti"].xcom_pull(key="next_color", task_ids="get_colors")
+
+    df = pd.read_csv(new_file, dtype=str)
+    connections.create_connection(
+        hosts=[ELASTIC_URL],
+        http_auth=(ELASTIC_USER, ELASTIC_PASSWORD),
+        retry_on_timeout=True,
+    )
+    elastic_connection = connections.get_connection()
+    list_errors = []
+    list_success = []
+
+    if type_file == "rge":
+        generations = generate_updates_rge(df, color)
+    if type_file == "spectacle":
+        generations = generate_updates_spectacle(df, color)
+    if type_file == "convcollective":
+        generations = generate_updates_convcollective(df, color)
+    if type_file == "uai":
+        generations = generate_updates_uai(df, color)
+    if type_file == "finess":
+        generations = generate_updates_finess(df, color)
+    if type_file == "colter":
+        generations = generate_updates_colter(df, color)
+    if type_file == "elu":
+        generations = generate_updates_elu(df, color)
```

Comment:
Maybe use a map here, instead of if conditions ?

Reply:
I let you manage that. I don't find satisfying way to do it. Do what you want!